data/account/participation_test.go

```diff
@@ -149,3 +172,279 @@ func BenchmarkOldKeysDeletion(b *testing.B) {
 	}
 	part.Close()
 }
+
+func TestRetrieveFromDB(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+	part, rootDB, partDB, err := setupParticipationKey(t, a)
+	a.NoError(err)
+	defer closeDBS(rootDB, partDB)
+
+	retrievedPart, err := RestoreParticipation(partDB)
+	a.NoError(err)
+	a.NotNil(retrievedPart)
+
+	// comparing the outputs:
+	a.Equal(intoComparable(part), intoComparable(retrievedPart))
+
+}
+
+func TestRetrieveFromDBAtVersion1(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	a := require.New(t)
+	ppart := setupkeyWithNoDBS(t, a)
+	_, rootDB, partDB := createTestDBs(a, t.Name())
+	defer closeDBS(rootDB, partDB)
+
+	part := ppart.Participation
+	a.NoError(setupTestDBAtVer1(partDB, part))
+
+	retrivedPart, err := RestoreParticipation(partDB)
+	a.NoError(err)
+	assertionForRestoringFromDBAtLowVersion(a, retrivedPart)
+}
+
+func TestRetriveFromDBAtVersion2(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	a := require.New(t)
+
+	ppart := setupkeyWithNoDBS(t, a)
+	_, rootDB, partDB := createTestDBs(a, t.Name())
+	defer closeDBS(rootDB, partDB)
+
+	part := ppart.Participation
+	a.NoError(setupTestDBAtVer2(partDB, part))
+
+	retrivedPart, err := RestoreParticipation(partDB)
+	a.NoError(err)
+	assertionForRestoringFromDBAtLowVersion(a, retrivedPart)
+}
+
+func TestKeyRegCreation(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	a := require.New(t)
+
+	ppart := setupkeyWithNoDBS(t, a)
+
+	cur := config.Consensus[protocol.ConsensusCurrentVersion]
+	txn := ppart.Participation.GenerateRegistrationTransaction(basics.MicroAlgos{Raw: 1000}, 0, 100, [32]byte{}, cur)
+	a.Equal(txn.StateProofPK.IsEmpty(), true)
+
+	future := config.Consensus[protocol.ConsensusFuture]
+	txn = ppart.Participation.GenerateRegistrationTransaction(basics.MicroAlgos{Raw: 1000}, 0, 100, [32]byte{}, future)
+	a.Equal(txn.StateProofPK.IsEmpty(), false)
```

Comment:
`Equal` first argument is the expected value, second is the checked value. The parameter order must be reversed.